pages/authors/[author].tsx

```diff
@@ -0,0 +1,38 @@
+import { GetStaticPaths, GetStaticProps } from 'next'
+
+import Author, { Props as AuthorProps } from '../../components/Author'
+import getQueryParameter from '../../util/getQueryParameters'
+import loadAllRecords from '../../util/loadAllRecords'
+import loadAuthorCollections from '../../util/loadAuthorCollections'
+import omitUndefinedFields from '../../util/omitUndefinedFields'
+
+export const getStaticPaths: GetStaticPaths = async () => {
+    const authorCollection =  await loadAuthorCollections()
+    const authors = authorCollection.authors.map(author => author.id)
+    const paths = authors.map(author => `/authors/${author}`)
```

Comment:
Just a small idea for the names of these variables. Since the values here are author IDs, you could name the variable to reflect this.
```suggestion
    const authorIds = authorCollection.authors.map(author => author.id)
    const paths = authorIds.map(authorId => `/authors/${authorId}`)
```